### DISCUSSION TITLE
Regex Help

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#65839] [Q&A] [answered]
I have inputs that looks like this:
```
This is an ABC-123 test.
This is an ABC - 123 test.
```

I am interested in capturing ```ABC-123``` (with no spaces in it) out of both pieces of text into a single piece of text or named capture.
Is that possible to do solely with regex or do I need to do named captures and combine them in code?

**Comment by stephentoub** [comment, marked as answer]
Questions like this are better for a site like stackoverflow.  The dotnet/runtime repo isn't intended to be a general Q&A destination for how to use .NET.  Thanks.

> Is that possible to do solely with regex or do I need to do named captures and combine them in code?

There's no mechanism in Regex to post-process captures as part of the match itself.  You can have a capture for the whole region, though, and then just remove the spaces yourself, e.g.
```C#
Regex r = new Regex(@"\w{3}\s*-\s*\d{3}"); // I'm assuming your pattern is something like this
...
Match m = r.Match(input);
while (m.Success)
{
    string valueWithSpaces = m.Value.Replace(" ", "");
    Use(valueWithSpaces);
    m = m.NextMatch();
}
```
or as you say have individual capture groups that you then combine:
```C#
Regex r = new Regex(@"(\w{3})\s*-\s*(\d{3})");
...
Match m = r.Match(input);
while (m.Success)
{
    string valueWithSpaces = string.Concat(m.Groups[1].ValueSpan, "-", m.Groups[2].ValueSpan);
    Use(valueWithSpaces);
    m = m.NextMatch();
}
```
